velox/exec/SortBuffer.cpp

```diff
@@ -47,9 +47,11 @@ SortBuffer::SortBuffer(
   sortedSpillColumnNames.reserve(input->size());
   sortedSpillColumnTypes.reserve(input->size());
   std::unordered_set<column_index_t> sortedChannelSet;
+  decodedInputs_.resize(input_->size());
   // Sorted key columns.
   for (column_index_t i = 0; i < sortColumnIndices.size(); ++i) {
     columnMap_.emplace_back(IdentityProjection(i, sortColumnIndices.at(i)));
+    decodedInputs_[sortColumnIndices.at(i)].second = i;
```

Comment:
Since the columnMap_ contain all the columns, the index is from 0 to numColumns, we can also change it to std::vector<int>, the index is the outoutChannel, the value in the index is the inputChannel, then we could not reserve  it in decodedInputs_.
We need change function `gatherCopy`  which is only used here too.

Reply:
Are you saying that you cannot create std::vector<DecodedVector> and then call resize(n) on it? I wonder why? DecodedVector has default constructor and supports std::move. What else is missing?